RFCs/0011-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,761 @@
+
+Optional attributes in target schema that have the same name in the resulting tuples from the source schema, SHOULD have the same type. If the implementing database provides type coercion rule for the types involved in source and destination, the `SELECT` MAY lead to a successful outcome, otherwise it's a `SemanticError`. See examples for more clarification.
+
+#### 3.4 Examples
+
+The following example statement inserts items from `Vehicles` table into `HeavyVehicles` table. Both tables have the same layout for required and optional attributes, therefore the `INSERT` statement can insert items using a `SELECT` query.
+
+```SQL
+-- The following `CREATE TABLE` is arbitrary since the PartiQL DDL is yet to be
+-- defined.CREATE TABLE Music
+-- (
+--    Artist     VARCHAR(20) NOT NULL,
+--    SongTitle  VARCHAR(30) NOT NULL,
+--    AlbumTitle VARCHAR(25) NOT NULL,
+--    Year       INT,
+--    Price      FLOAT,
+--    Genre      VARCHAR(10),
+--    PRIMARY KEY (Artist, SongTitle)
+-- );
+
+CREATE TABLE RockAlbums
+(
+    Artist     VARCHAR(20) NOT NULL,
+    SongTitle  VARCHAR(30) NOT NULL,
+    AlbumTitle VARCHAR(25) NOT NULL,
+    Year       INT,
+    RockGenre  VARCHAR(10),
+    PRIMARY KEY (Artist, SongTitle)
+);
+
+INSERT INTO Music
+SELECT *
+FROM RockAlbums
+WHERE RockGenre IN ('Alternative', 'SpaceRock');
+```
+
+The following statement shows that the values can be specified with a sub-select.
+
+```SQL
+INSERT INTO Music (Artist, SongTitle)
+SELECT Artist, SongTitle
+FROM RockAlbums
+WHERE RockGenre IN ('Alternative', 'SpaceRock');
+```
+
+The following example statement attempts to insert items from `Vehicles` table to `HeavyVehicles` table. Both tables have the same layout for required attributes but have different types for `Year` optional attribute, therefore—if the implementing database system does not implement a type coercion rule from `DATE` to `INT`—the `INSERT` statement using a `SELECT` query leads to a `SemanticError`.
+
+```SQL
+-- The following `CREATE TABLE` is arbitrary since the PartiQL DDL is yet to be-- defined.CREATE TABLE Music
+-- (
+--    Artist     VARCHAR(20) NOT NULL,
+--    SongTitle  VARCHAR(30) NOT NULL,
+--    AlbumTitle VARCHAR(25) NOT NULL,
+--    Year       INT,
+--    Price      FLOAT,
+--    Genre      VARCHAR(10),
+--    PRIMARY KEY (Artist, SongTitle)
+-- );
+
+CREATE TABLE RockAlbums
+(
+    Artist     VARCHAR(20) NOT NULL,
+    SongTitle  VARCHAR(30) NOT NULL,
+    AlbumTitle VARCHAR(25) NOT NULL,
+    Year       DATE,
+    RockGenre  VARCHAR(10),
+    PRIMARY KEY (Artist, SongTitle)
+);
+
+INSERT INTO MusicSELECT * FROM RockAlbums;
+```
+
+The following example statement, inserts some rows into table `films` from a table `tmp_films`. Both tables must have the same layout for required and optional attributes that are in common.
```

Comment:
There's some ambiguity here:  what about attributes that are *not* in common?  Is that even allowed?

Reply:
Yes I agree. It requires more explanation; in that case, it depends on the target schema. If it's an open schema, then yes, otherwise, no.